### ISSUE TITLE
Clarify `instanceNormalization`'s support for 1d, 2d, 3d use cases

### ISSUE THREAD

**Issue by huningxin** [webmachinelearning/webnn#490]
(This was raised by @wacky6 in Chromium [CL-5092348](https://chromium-review.googlesource.com/c/chromium/src/+/5092348/comment/bfa269c7_d3844d3a/) review.)

WebNN's [`instanceNormalization` ](https://www.w3.org/TR/webnn/#api-mlgraphbuilder-instancenorm) accepts 4-D input:

> input: an [MLOperand](https://www.w3.org/TR/webnn/#mloperand). The input 4-D tensor.

Jiewei proposed to rename it to `instanceNormalization2d` that clarify its 2d data use case.

> In this case, i would recommend defining IDL as instanceNormalization2d. Looks like the current definition is only supposed to work on 2d data.

And Jiewei also suggested PyTorch has separate [`InstanceNorm1d`](https://pytorch.org/docs/stable/generated/torch.nn.InstanceNorm1d.html#torch.nn.InstanceNorm1d), [`InstanceNorm2d`](https://pytorch.org/docs/stable/generated/torch.nn.InstanceNorm2d.html#torch.nn.InstanceNorm2d), [`InstanceNorm3d`](https://pytorch.org/docs/stable/generated/torch.nn.InstanceNorm3d.html#torch.nn.InstanceNorm3d) for different use cases accordingly

/cc @fdwr @wchao1115

**Comment by fdwr** [comment]
> And Jiewei also suggested PyTorch has separate [InstanceNorm1d](https://pytorch.org/docs/stable/generated/torch.nn.InstanceNorm1d.html#torch.nn.InstanceNorm1d), [InstanceNorm2d](https://pytorch.org/docs/stable/generated/torch.nn.InstanceNorm2d.html#torch.nn.InstanceNorm2d), [InstanceNorm3d](https://pytorch.org/docs/stable/generated/torch.nn.InstanceNorm3d.html#torch.nn.InstanceNorm3d)

Add 3 separate methods that basically all do the same thing except for operating on different axes (`[2]` / `[2,3]` / `[2,3,4]`)? 😮 I know WebNN did that for conv\*2d/pool\*2d/resample2d already, but if we continue appending suffixes to each operator for each possible rank, our operator list will become rather prolific. It would also become inconsistent with naming for the other two normalization functions.

### Counterproposal
Delete `instanceNormalization` entirely, as `layerNormalization` in Chai's spec obviates it with superset functionality. That tightens the WebNN API, reduces test and implementation costs, and avoids another round of debates on tensor layouts by removing the now unnecessary [`MLInputOperandLayout`](https://www.w3.org/TR/webnn/#enumdef-mlinputoperandlayout) field, since dimensions are just dimensions.

### Evidence LN is a superset of IN
There is a direct and trivial mapping:

```
instanceNormalization(i, {"scale": s, "bias": b, "epsilon": e, "layout": "nchw"} ->
   layerNormalization(i, {"scale": s, "bias": b, "epsilon": e, "axes"  : [2,3] }

instanceNormalization(i, {"scale": s, "bias": b, "epsilon": e, "layout": "nhwc"} ->
   layerNormalization(i, {"scale": s, "bias": b, "epsilon": e, "axes"  : [1,2] }
```

<table><tr><td>

```javascript
dictionary MLInstanceNormalizationOptions {
  MLOperand scale;
  MLOperand bias;
  float epsilon = 1e-5
  MLInputOperandLayout layout = "nchw"
}
```
</td>
<td>

```javascript
dictionary MLLayerNormalizationOptions {
  MLOperand scale;
  MLOperand bias;
  sequence<unsigned long> axes;
  float epsilon = 1e-5;
};
```
</tr></table>

### Backend considerations

If a backend has an optimized implementation that only accepts axes [2,3] (consistent with instanceNorm behavior) which would be faster to call than it's general `layerNormalization` implementation, then it's just a check (`if (input.rank == 4 && axes == [2,3]) backendNormalizationFunctionOptimizedHW(...); else backendNormalizationFunctionGeneral(...)`).

**Comment by Yaffle** [comment]
`resample2d` ends with `2d`, while works for any rank, which is inconsistent with other "2d" ops
`instanceNormalization` does not end with `2d`, but works only for nchw/nhwc shapes

**Comment by fdwr** [comment]
> `resample2d` ends with `2d`, while works for any rank

@Yaffle : `resample2d` currently expects a 4D input tensor, and only resamples along 2 dimensions ([`axes ... 
A list of length 2. Specifies the two dimensions of the input tensor t`](https://webmachinelearning.github.io/webnn/#api-mlgraphbuilder-resample2d-method)), hence the "2D", but there's [another issue](https://github.com/webmachinelearning/webnn/issues/904) to generalize/relax this now unnecessary rank restriction for a few ops, which would rename the bunch. Maybe I should include the `normalize*` ops (`normalizeInstance`, `normalizeBatch`, `normalizeLayer`) in that issue too 🤔...
